### PR TITLE
Fix dynamic sliding-window KV cache mask for multi-token causal decode

### PR DESCRIPTION
**Pull request by hikettei**
Fixed a bug in DynamicKVCacheLayer.attention_mask() for causal attention with sliding_window_size. Previously, the dynamic KV cache built the sliding-window mask with tril/triu based on suffix-local row indices. That was incorrect when a prefix KV cache was already present, because the visible window should be defined in terms of absolute query/key positions.

As a result, for models using sliding-window attention:
- single-token sequential decode
- multi-token causal decode

could attend to different key sets, producing deterministic hidden-state / logit mismatches.

### CODE REVIEW

File: lalamo/modules/token_mixers/state/kv_cache.py [hunk 1]
```diff
@@ -89,16 +89,26 @@ def attention_mask(
...
+            current_length = (
+                self.keys.shape[0] if self.padding_mask is None else jnp.sum(self.padding_mask, dtype=jnp.int32)
+            )
+            query_indices = current_length + query_offsets
+            key_indices = jnp.arange(total_num_tokens, dtype=jnp.int32)
+
+            result = query_indices[:, None] >= key_indices[None, :]
             if sliding_window_size is not None:
-                result = jnp.triu(result, k=1 - sliding_window_size)
+                result = result & (query_indices[:, None] < (key_indices[None, :] + sliding_window_size))
```
Comment (knyazer) [inline lalamo/modules/token_mixers/state/kv_cache.py]: let's use jnp.logical_and instead of &, lgtm otherwise

**Review by knyazer — APPROVED** [review]

**Review by knyazer — CHANGES_REQUESTED** [review]
wait sorry, one thing - test_sliding_window_mask should be faster, right now the aim of all the tests in unit/ to be ultra fast, and it takes 8 seconds on my machine

I'll also run the longer test suite to make sure we don't break any models by accident

File: lalamo/modules/token_mixers/state/kv_cache.py [hunk 2]
```diff
@@ -89,16 +89,26 @@ def attention_mask(
...
+        if suffix_length_without_padding is None:
+            suffix_length_without_padding = suffix_length
+
         result = jnp.ones((suffix_length, total_num_tokens), dtype=jnp.bool)
         if is_causal:
-            result = jnp.tril(result, k=total_num_tokens - suffix_length)
+            query_offsets = jnp.arange(0, suffix_length, dtype=jnp.int32) - suffix_length_without_padding
+            current_length = (
+                self.keys.shape[0] if self.padding_mask is None else jnp.sum(self.padding_mask, dtype=jnp.int32)
+            )
```
[the remark this answers is not in the document]
Comment (knyazer) [reply lalamo/modules/token_mixers/state/kv_cache.py]: also please use shape unpacking instead of direct indexing

**Review by norpadon — APPROVED** [review]

**Review by knyazer — APPROVED** [review]
good job
